pkg/options/types.go

```diff
@@ -105,14 +106,52 @@ func (r *ResourceSet) Type() string {
 }
 
 // NodeType represents a nodeName to query from.
-type NodeType string
+type NodeType map[string]struct{}
+
+func (n *NodeType) Set(value string) error {
+	s := *n
+	cols := strings.Split(value, ",")
+	for _, col := range cols {
+		col = strings.TrimSpace(col)
+		if len(col) != 0 {
+			s[col] = struct{}{}
+		}
+	}
+	return nil
+}
+
+func (n NodeType) AsSlice() []string {
+	cols := make([]string, 0, len(n))
+	for col := range n {
+		cols = append(cols, col)
+	}
+	return cols
+}
+
+func (n NodeType) String() string {
+	return strings.Join(n.AsSlice(), ",")
+}
+
+func (n *NodeType) Type() string {
+	return "string"
+}
 
 // GetNodeFieldSelector returns a nodename field selector.
 func (n *NodeType) GetNodeFieldSelector() string {
-	if string(*n) != "" {
-		return fields.OneTermEqualSelector("spec.nodeName", string(*n)).String()
+	if nil == n || len(*n) == 0 {
+		klog.InfoS("Using node type is nil")
+		return EmptyFieldSelector()
 	}
-	return EmptyFieldSelector()
+	pattern := "[^a-zA-Z0-9_,-]+"
+	re := regexp.MustCompile(pattern)
+	result := re.ReplaceAllString(n.String(), "")
```

Comment:
QQ: why do we need `re.ReplaceAllString(n.String(), "")`

Reply:
nit: remove `re.ReplaceAllString(n.String(), "")`. If node name isn't correct, keep it as before. Then users can know node name isn't correct.

Reply:
> QQ: why we need `re.ReplaceAllString(n.String(), "")`

There are two considerations for regularization here: 
1. Handle the difference between "" and nil 
2. Correction of node name

Reply:
QQ: could you give two examples for these two cases?

We shouldn't change node name, which users are not aware of.

Reply:
Ping~

Keep node name as same from input.

Reply:
@mickeyzzc can you address this comment?